cfgov/teachers_digital_platform/tests/models/test_activity_index_page.py

```diff
@@ -60,51 +66,282 @@ def test_can_create_activity_index_page(self):
+        mock_response = mock.Mock(dsl_response)
+        mock_response.__iter__ = mock.Mock(return_value=iter([mock_hit]))
+        mock_response.aggregations.topic_terms.buckets\
+            .return_value = mock.Mock(
+                return_value=iter({
+                    'topic': [{'key': '1'}]
+                })
+            )
+        mock_search().from_dict().query().count = mock.Mock(return_value=1)
+        mock_search().from_dict().query()\
+            .__getitem__().execute.return_value = mock_response
+        response = self.client.get(f"{self.search_page.url}?q=test-query")
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(
+            "test-query",
+            response.content.decode('utf8')
+        )
+
+    @override_settings(FLAGS={"ELASTICSEARCH_DSL_TDP": [("boolean", True)]})
+    @mock.patch.object(ActivityPageDocument, 'search')
+    def test_search_page_renders_with_facet_parameters(self, mock_search):
+        mock_hit = mock.Mock(Hit)
+        mock_hit.id = self.activity_page.pk
+        mock_response = mock.Mock(dsl_response)
+        mock_response.__iter__ = mock.Mock(return_value=iter([mock_hit]))
+        mock_response.aggregations.topic_terms.buckets\
+            .return_value = mock.Mock(
+                return_value=iter({
+                    'topic': [{'key': '1'}, {'key': '2'}, {'key': '3'}]
+                })
+            )
+        mock_search().from_dict().sort().query().count = mock.Mock(
+            return_value=1
+        )
+        mock_search().from_dict().sort().query()\
+            .__getitem__().execute.return_value = mock_response
+        response = self.client.get(
+            f"{self.search_page.url}?topic=1&topic=2&topic=3")
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(
+            "topic=1",
+            response.content.decode('utf8')
+        )
+
+    def test_taxonomy_model_str(self):
+        taxonomy_instance = ActivityBuildingBlock.objects.first()
+        self.assertEqual(
+            str(taxonomy_instance),
+            taxonomy_instance.title
+        )
+
+    def test_mptt_model_str(self):
+        mptt_instance = ActivityTopic.objects.first()
+        self.assertEqual(
+            str(mptt_instance),
+            mptt_instance.title
+        )
+
+    def test_validate_pagination_number(self):
+        paginator = Paginator([{"fake": "results"}] * 30, 25)
+        request = HttpRequest()
+        self.assertEqual(validate_page_number(request, paginator), 1)
+        request.GET.update({"page": "2"})
+        self.assertEqual(validate_page_number(request, paginator), 2)
+        request = HttpRequest()
+        request.GET.update({"page": "1000"})
+        self.assertEqual(validate_page_number(request, paginator), 1)
+        request = HttpRequest()
+        request.GET.update({"page": "<script>Boo</script>"})
+        self.assertEqual(validate_page_number(request, paginator), 1)
```

Comment:
This test (and the function tested) exist in both ask_cfpb and teachers_digital_platform.
We should (in another PR) move the function and test to the search app for sharing.

Reply:
There might also be a search function or a test in paying_for_college that exist in an other elasticsearch application.
To reduce code any common search functions or tests should be moved to the search application for sharing in that PR.